stress/shell/src/main/java/alluxio/stress/cli/StressJobServiceBench.java

```diff
@@ -238,40 +259,106 @@ private void applyOperation(String dirPath) throws IOException, AlluxioException
-        }
-        os.write(arr);
+    private void runDistributedLoad(String dirPath) throws AlluxioException, IOException {
+      int numReplication = 1;
+      DistributedLoadCommand cmd = new DistributedLoadCommand(mFsContext);
+      try {
+        DistributedLoadUtils.distributedLoad(cmd, new AlluxioURI(dirPath), numReplication,
+            new HashSet<>(), new HashSet<>(), new HashSet<>(), new HashSet<>(), false);
+      } finally {
+        mResult.incrementNumSuccess(cmd.getCompletedCount());
       }
     }
+  }
+
+  private void createFiles(FileSystem fs, int numFiles, String dirPath, int fileSize)
+      throws IOException, AlluxioException {
+    CreateFilePOptions options =
+        CreateFilePOptions.newBuilder().setRecursive(true).setWriteType(WritePType.THROUGH).build();
 
-    private void deletePath(FileSystem fs, String dirPath) throws IOException, AlluxioException {
-      AlluxioURI path = new AlluxioURI(dirPath);
-      if (fs.exists(path)) {
-        DeletePOptions options = DeletePOptions.newBuilder().setRecursive(true).build();
-        fs.delete(path, options);
+    for (int fileId = 0; fileId < numFiles; fileId++) {
+      String filePath = String.format("%s/%d", dirPath, fileId);
+      createByteFile(fs, new AlluxioURI(filePath), options, fileSize);
+    }
+  }
+
+  private void createByteFile(FileSystem fs, AlluxioURI fileURI, CreateFilePOptions options,
+      int len) throws IOException, AlluxioException {
+    try (FileOutStream os = fs.createFile(fileURI, options)) {
+      byte[] arr = new byte[len];
+      for (int k = 0; k < len; k++) {
+        arr[k] = (byte) k;
       }
+      os.write(arr);
+    }
+  }
+
+  private void deletePath(FileSystem fs, String dirPath) throws IOException, AlluxioException {
+    AlluxioURI path = new AlluxioURI(dirPath);
+    if (fs.exists(path)) {
+      DeletePOptions options = DeletePOptions.newBuilder().setRecursive(true).build();
+      fs.delete(path, options);
+    }
+  }
+
+  private void runNoop() throws IOException, InterruptedException, TimeoutException {
+    long jobId = mJobMasterClient.run(new NoopPlanConfig());
+    // TODO(jianjian): refactor JobTestUtils
+    ImmutableSet<Status> statuses =
+        ImmutableSet.of(Status.COMPLETED, Status.CANCELED, Status.FAILED);
+    final AtomicReference<JobInfo> singleton = new AtomicReference<>();
+    CommonUtils.waitFor(
+        String.format("job %d to be one of status %s", jobId, Arrays.toString(statuses.toArray())),
+        () -> {
+          JobInfo info;
+          try {
+            info = mJobMasterClient.getJobStatus(jobId);
+            if (statuses.contains(info.getStatus())) {
+              singleton.set(info);
+            }
+            return statuses.contains(info.getStatus());
+          } catch (IOException e) {
+            throw Throwables.propagate(e);
+          }
+        }, WaitForOptions.defaults().setTimeoutMs(30 * Constants.SECOND_MS));
+    JobInfo jobInfo = singleton.get();
```

Comment:
is the reason for using an AtomicReference here because the waitfor signature does not allow return values? what if the reference is never set (e.g. an exception was thrown)?

Reply:
yes, I'm using the same function as JobTestUtils(think of refactoring them in the following PR). I think if any exception happened during the waiting time, It would propagate the exception and would try to catch that exception in line 215.